### DISCUSSION TITLE
Should I use Payload for Nextjs? Performance problem

### DISCUSSION THREAD

**Discussion by IRediTOTO** [payloadcms/payload#240] [Q&A]
Hi, I just read this post https://payloadcms.com/blog/nextjs-payload-typescript-single-express-server-boilerplate
But Nextjs said:`A custom server will remove important performance optimizations, like serverless functions and Automatic Static Optimization.`
Can you please comfirm?

**Comment by richardvanbergen** [comment]
Yes it will but you don't have to use payload like that if you don't want to you. You can install it separately and run next and payload independently of each other.

https://payloadcms.com/docs/getting-started/installation

Then you just use next functions like `getServerSideProps` and you'll get access to automatic static rendering.

It is a bit harder to setup and use that way though so it depends on how badly you want those nextjs features.

**Reply by IRediTOTO** [reply]
Dont you have tutorials deploy on heroku?

**Reply by Ontopic** [reply]
No automatic demo deploys yet. Feel it would be a waste anyway, the power of Payload becomes visible if you dive into it. The issue of running Payload and NextJS separately is not that difficult. I would recommend however to not use the local API anywhere if you go for separating them, but from the start rely on API calls.

Actually I feel there are more "traps" combining Next and Payload in one process. Importing files you accidentally into the "wrong" layer becomes a little too easy. I personally went with a setup that has 3 code layers, client (nextjs), bridge (shared components and connections with sql database and mongo directly [allowing client to keep running without Payload if needed or for quick dev builds]) and backend (payload).

But as said, I think the beauty in Payload is the flexibility to take your own approach. So go play around and see what feels right to you :)

**Reply by liamrohan** [reply]
> It is a bit harder to setup and use that way though so it depends on how badly you want those nextjs features.

Why would it be harder? Wouldn't it be easier to separate the two? 

As @Ontopic mentioned essentially relying on payload as an easy way to interact with MongoDB and speed up API dev. Would be great if there was an example like `payload-nextjs-server` showing this.

**Reply by richardvanbergen** [reply]
I was referring to the deployment/publish/build process for statically generated sites. You need two build/deployment processes and you need a way of content editors being able to publish content they edit.

Generally speaking I find it easier to just have one deployment and pull the data are runtime but that obviously results in slower responses.

**Reply by philval** [reply]
Hi. It would be good to see some Architecture diagrams of alternative setups on the PCMS website ( with trade-offs... ) and possible / recommended hosting options for each "box", "layer" etc.

**Reply by jmikrut** [reply]
Hey @philval — this is a great suggestion. 

We're actually working on building Payload Cloud, which will be by far the best way to deploy a Payload app (complete with database and permanent S3-based file storage). 

This will influence our architecture diagrams but we will be spending some more time here soon.

Thanks for the suggestion!

**Reply by liamrohan** [reply]
@jmikrut Any hints on a rough release date? Thanks

**Reply by jmikrut** [reply]
Not yet, but what I can say is that our first priority is hitting 1.0, which will be happening within the next month. Then after that, all our efforts are going to switch to building Payload Cloud. We've got a pretty good vision so it should move fairly swiftly, but if you've got ideas or input as to what you'd like to see, we'd love to hear. Swing over to the [Discussion](https://github.com/payloadcms/payload/discussions/583) and leave some thoughts if you have any!
